pkg/controller/datavolume/util.go

```diff
@@ -406,15 +404,14 @@ func getDefaultVolumeMode(c client.Client, storageClass *storagev1.StorageClass,
 				}
 			}
 		}
-		// if not found return default volume mode for the storage class
-		return volumeMode, nil
 	}
 
-	// since volumeMode is optional - > gracefully fallback to k8s defaults,
+	// since volumeMode is optional, gracefully fallback to k8s defaults when no matching accessMode
```

Comment:
Should this be `when no matching volumeMode`?

Reply:
right, fixed.